cf/api/service_brokers.go

```diff
@@ -56,7 +56,7 @@ func (repo CloudControllerServiceBrokerRepository) FindByName(name string) (serv
 			return false
 		})
 
-	if !foundBroker {
+	if !foundBroker && (apiErr == nil) {
```

Comment:
I think the test coverage is sufficient; there's no case where we'd have a list of service brokers to search as well as an apiErr from CC. If we get a CC error, we return from ListPaginatedResources before ever searching for a potential service broker. foundBroker will always remain false if there's a CC error.

This block covers a weird case: the CC may return 200 with an empty list, so we make our own error in this if-block. The additional check is a bug fix; the previous behavior was ignoring existing CC errors and would overwrite them with the ModelNotFoundError.